pkg/virt-controller/services/template.go

```diff
@@ -396,872 +396,327 @@ func generateQemuTimeoutWithJitter(qemuTimeoutBaseSeconds int) string {
-			sizeLimit := resource.MustParse("1Mi")
-			volumes = append(volumes, k8sv1.Volume{
-				Name: volume.Name,
-				VolumeSource: k8sv1.VolumeSource{
-					EmptyDir: &k8sv1.EmptyDirVolumeSource{
-						Medium:    "Memory",
-						SizeLimit: &sizeLimit,
-					},
-				},
-			})
-			volumeMounts = append(volumeMounts, k8sv1.VolumeMount{
-				Name:      volume.Name,
-				MountPath: config.DownwardMetricDisksDir,
-			})
-		}
+	// Add ports from interfaces to the pod manifest
+	ports := getPortsFromVMI(vmi)
 
-		if volume.CloudInitNoCloud != nil {
-			if volume.CloudInitNoCloud.UserDataSecretRef != nil {
-				// attach a secret referenced by the user
-				volumeName := volume.Name + "-udata"
-				volumes = append(volumes, k8sv1.Volume{
-					Name: volumeName,
-					VolumeSource: k8sv1.VolumeSource{
-						Secret: &k8sv1.SecretVolumeSource{
-							SecretName: volume.CloudInitNoCloud.UserDataSecretRef.Name,
-						},
-					},
-				})
-				volumeMounts = append(volumeMounts, k8sv1.VolumeMount{
-					Name:      volumeName,
-					MountPath: filepath.Join(config.SecretSourceDir, volume.Name, "userdata"),
-					SubPath:   "userdata",
-					ReadOnly:  true,
-				})
-				volumeMounts = append(volumeMounts, k8sv1.VolumeMount{
-					Name:      volumeName,
-					MountPath: filepath.Join(config.SecretSourceDir, volume.Name, "userData"),
-					SubPath:   "userData",
-					ReadOnly:  true,
-				})
-			}
-			if volume.CloudInitNoCloud.NetworkDataSecretRef != nil {
-				// attach a secret referenced by the networkdata
-				volumeName := volume.Name + "-ndata"
-				volumes = append(volumes, k8sv1.Volume{
-					Name: volumeName,
-					VolumeSource: k8sv1.VolumeSource{
-						Secret: &k8sv1.SecretVolumeSource{
-							SecretName: volume.CloudInitNoCloud.NetworkDataSecretRef.Name,
-						},
-					},
-				})
-				volumeMounts = append(volumeMounts, k8sv1.VolumeMount{
-					Name:      volumeName,
-					MountPath: filepath.Join(config.SecretSourceDir, volume.Name, "networkdata"),
-					SubPath:   "networkdata",
-					ReadOnly:  true,
-				})
-				volumeMounts = append(volumeMounts, k8sv1.VolumeMount{
-					Name:      volumeName,
-					MountPath: filepath.Join(config.SecretSourceDir, volume.Name, "networkData"),
-					SubPath:   "networkData",
-					ReadOnly:  true,
-				})
-			}
-		}
+	capabilities := getRequiredCapabilities(vmi, t.clusterConfig)
 
-		if volume.Sysprep != nil {
-			var volumeSource k8sv1.VolumeSource
-			// attach a Secret or ConfigMap referenced by the user
-			volumeSource, err := sysprepVolumeSource(*volume.Sysprep)
-			if err != nil {
-				return nil, err
-			}
-			volumes = append(volumes, k8sv1.Volume{
-				Name:         volume.Name,
-				VolumeSource: volumeSource,
-			})
-			volumeMounts = append(volumeMounts, k8sv1.VolumeMount{
-				Name:      volume.Name,
-				MountPath: filepath.Join(config.SysprepSourceDir, volume.Name),
-				ReadOnly:  true,
-			})
-		}
+	networkToResourceMap, err := getNetworkToResourceMap(t.virtClient, vmi)
```

Comment:
Can't this whole snippet 
```go
networkToResourceMap, err := getNetworkToResourceMap(t.virtClient, vmi)
	if err != nil {
		return nil, err
	}

	// Register resource requests and limits corresponding to attached multus networks.
	for _, resourceName := range networkToResourceMap {
		if resourceName != "" {
			requestResource(&resources, resourceName)
		}
	}

	err = validatePermittedHostDevices(&vmi.Spec, t.clusterConfig)
	if err != nil {
		return nil, err
	}

	if util.IsGPUVMI(vmi) {
		for _, gpu := range vmi.Spec.Domain.Devices.GPUs {
			requestResource(&resources, gpu.DeviceName)
		}
	}

	if util.IsHostDevVMI(vmi) {
		for _, hostDev := range vmi.Spec.Domain.Devices.HostDevices {
			requestResource(&resources, hostDev.DeviceName)
		}
	}
```
be included in `renderLaunchManifestResourceRequirements` ? We're always dealing with resources, I think it makes sense to group them all under the same function.